cleverhans/experimental/certification/dual_formulation.py

```diff
@@ -303,31 +357,88 @@ def get_full_psd_matrix(self):
         axis=0)
     return self.matrix_h, self.matrix_m
 
-  def compute_certificate(self):
+  def make_M_psd(self):
+    """Run binary search to find a value for nu that makes M PSD
+
+    Args:
+      feed_dict: dictionary of updated lambda variables to feed into M
+    Returns:
+      new_nu: new value of nu
+    """
+    original_nu = self.sess.run(self.nu)
+    _, min_eig_val_m = self.get_lanczos_eig()
+
+    lower_nu = original_nu
+    upper_nu = original_nu
+    num_iter = 0
+
+    # Find an upper bound on nu
+    while min_eig_val_m - TOL < 0:
+      if num_iter >= 15:
+        break
+      num_iter += 1
+      upper_nu *= 1.1
+      self.sess.run(tf.assign(self.nu, upper_nu))
+      _, min_eig_val_m = self.get_lanczos_eig()
+
+    final_nu = upper_nu
+
+    # Perform binary search to find best value of nu
+    while lower_nu <= upper_nu:
+      if num_iter >= 25:
+        final_nu = upper_nu
+        self.sess.run(tf.assign(self.nu, final_nu))
+        break
+      num_iter += 1
+      mid_nu = (lower_nu + upper_nu) / 2
+      self.sess.run(tf.assign(self.nu, mid_nu))
+      _, min_eig_val_m = self.get_lanczos_eig()
+      if min_eig_val_m - TOL < 0:
+        lower_nu = mid_nu
+      else:
+        upper_nu = mid_nu
+
+    # Add 0.05 to final nu to account for numerical instability
+    return original_nu, final_nu + 0.05
+
+  def get_lanczos_eig(self, M=True):
```

Comment:
what is M and why default to true?
Please add description of the argument to the function comment.

Also name of the argument is very confusing, please rename it.
By looking only on the name, one can think that M is supposed to be value of the matrix, however it's apparently not. Generally you should try to make variable/argument names to be self explanatory

Also it's recommended to name variables and arguments using `lowercase_with_underscore`
See python style guide about naming: https://www.python.org/dev/peps/pep-0008/#function-and-variable-names